In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np


In [4]:
df = pd.read_csv('/content/drive/MyDrive/datasets/E-commerce Customer Behavior - Sheet1.csv')
df.head()

Customer ID  Gender  Age           City Membership Type  Total Spend  \
0          101  Female   29       New York            Gold      1120.20   
1          102    Male   34    Los Angeles          Silver       780.50   
2          103  Female   43        Chicago          Bronze       510.75   
3          104    Male   30  San Francisco            Gold      1480.30   
4          105    Male   27          Miami          Silver       720.40   

   Items Purchased  Average Rating  Discount Applied  \
0               14             4.6              True   
1               11             4.1             False   
2                9             3.4              True   
3               19             4.7             False   
4               13             4.0              True   

   Days Since Last Purchase Satisfaction Level  
0                        25          Satisfied  
1                        18            Neutral  
2                        42        Unsatisfied  
3                        12          Satisfied  
4                        55        Unsatisfied

In [5]:
df.shape

(350, 11)


**First choice: dataset 1**
E-commerce Customer Behavior Dataset




1. Source of data

Kaggle =

https://www.kaggle.com/datasets/uom190346a/e-commerce-customer-behavior-dataset

2. Brief description of data


This dataset provides a comprehensive view of customer behavior within an e-commerce platform. Each entry in the dataset corresponds to a unique customer, offering a detailed breakdown of their interactions and transactions. The information is crafted to facilitate a nuanced analysis of customer preferences, engagement patterns, and satisfaction levels, aiding businesses in making data-driven decisions to enhance the customer experience.

3. What is the target?

The Target is Satisfaction Level

4. What does one row represent? (A person? A business? An event? A product?)

A person and his interaction

5. Is this a classification or regression problem?

it's a classification problem

6. How many features does the data have?
11 Features
7. How many rows are in the dataset? 350 rows

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?



 Building predictive models from the dataset could pose challenges. For instance, selecting appropriate algorithms to predict customer behavior accurately, dealing with imbalanced data (if any), or ensuring the model's generalizability might be complex tasks.





In [6]:
df2 = pd.read_csv('/content/drive/MyDrive/datasets/Car_Insurance_Claim.csv')
df2.head()

ID    AGE  GENDER      RACE DRIVING_EXPERIENCE    EDUCATION  \
0  569520    65+  female  majority               0-9y  high school   
1  750365  16-25    male  majority               0-9y         none   
2  199901  16-25  female  majority               0-9y  high school   
3  478866  16-25    male  majority               0-9y   university   
4  731664  26-39    male  majority             10-19y         none   

          INCOME  CREDIT_SCORE  VEHICLE_OWNERSHIP VEHICLE_YEAR  MARRIED  \
0    upper class      0.629027                1.0   after 2015      0.0   
1        poverty      0.357757                0.0  before 2015      0.0   
2  working class      0.493146                1.0  before 2015      0.0   
3  working class      0.206013                1.0  before 2015      0.0   
4  working class      0.388366                1.0  before 2015      0.0   

   CHILDREN  POSTAL_CODE  ANNUAL_MILEAGE VEHICLE_TYPE  SPEEDING_VIOLATIONS  \
0       1.0        10238         12000.0        sedan                    0   
1       0.0        10238         16000.0        sedan                    0   
2       0.0        10238         11000.0        sedan                    0   
3       1.0        32765         11000.0        sedan                    0   
4       0.0        32765         12000.0        sedan                    2   

   DUIS  PAST_ACCIDENTS  OUTCOME  
0     0               0      0.0  
1     0               0      1.0  
2     0               0      0.0  
3     0               0      0.0  
4     0               1      1.0

In [7]:
df2.shape

(10000, 19)

1. Source of data

Kaggle =

https://www.kaggle.com/datasets/sagnik1511/car-insurance-data

2. Brief description of data


The company has shared its annual car insurance data. Now, you have to find out the real customer behaviors over the data.


3. What is the target?

The Target is The outcome column

4. What does one row represent? (A person? A business? An event? A product?)

A person and his interaction

5. Is this a classification or regression problem?

it's a classification problem

6. How many features does the data have?
18 Features
7. How many rows are in the dataset? 10 000  rows



8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

Data Cleaning: The dataset might have missing values, inconsistencies, or errors that need to be addressed before analysis.
